pages/_document.tsx

```diff
@@ -13,7 +12,7 @@ export default function Document() {
       <body>
         <Main />
         <NextScript />
-        <Script strategy="beforeInteractive" src="/static/js/legacyMain.js" />
+        <script src="/static/js/legacyMain.js" defer async />
```

Comment:
Why removing the `Script` tag from `nextjs`? Which kind o tests you made that resulted in this conclusion?

Reply:
This reason is hard to explain....But after a period of experienment, I found if I changed to this, it seems the bug was fixed... You can have a try after applying it on your local code. And That's the reason why I thought of it.

Any other better suggestions is welcomed.

Reply:
Maybe the customized event handlers' problem for listening after all the other js files of next.js are loaded successfully for querySelectoAll()……I also wonder if it's a better solution？

Reply:
Not really, it's more like the following:

1. The next/link component takes an href argument, which is pretty much what our JavaScript legacy script does, replace the URL
2. But the thing is that `next/link` also sets up "click" events and other kinds of events. These events are managed by a React component, meaning if there's a re-render or whatever the internal logic is, it will ignore our "href" attribute.
3. Not to mention, "click" event has precedence over "href". So basically, us replacing the "href" probably has no effect at that point in time. Note that I am just making a wild guess here.

But the actual solution is for now to not use next/link, we probably also should not use it for external links, as it was primarily made for internal links.